src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,471 @@
+        {
+            PseudoClasses.Set(pcFlyoutOpen, _isFlyoutOpen);
+            PseudoClasses.Set(pcChecked, GetValue(ToggleSplitButton.IsCheckedProperty));
+        }
+
+        /// <summary>
+        /// Opens the secondary button's flyout.
+        /// </summary>
+        protected void OpenFlyout()
+        {
+            if (Flyout != null)
+            {
+                Flyout.ShowAt(this);
+            }
+        }
+
+        /// <summary>
+        /// Closes the secondary button's flyout.
+        /// </summary>
+        protected void CloseFlyout()
+        {
+            if (Flyout != null)
+            {
+                Flyout.Hide();
+            }
+        }
+
+        /// <summary>
+        /// Registers all flyout events.
+        /// </summary>
+        /// <param name="flyout">The flyout to connect events to.</param>
+        private void RegisterFlyoutEvents(FlyoutBase flyout)
+        {
+            if (flyout != null)
+            {
+                flyout.Opened += Flyout_Opened;
+                flyout.Closed += Flyout_Closed;
+
+                _flyoutPropertyChangedDisposable = flyout.GetPropertyChangedObservable(FlyoutBase.PlacementProperty).Subscribe(Flyout_PlacementPropertyChanged);
+            }
+        }
+
+        /// <summary>
+        /// Explicitly unregisters all flyout events.
+        /// </summary>
+        /// <param name="flyout">The flyout to disconnect events from.</param>
+        private void UnregisterFlyoutEvents(FlyoutBase flyout)
+        {
+            if (flyout != null)
+            {
+                flyout.Opened -= Flyout_Opened;
+                flyout.Closed -= Flyout_Closed;
+
+                _flyoutPropertyChangedDisposable?.Dispose();
+                _flyoutPropertyChangedDisposable = null;
+             }
+        }
+
+        /// <summary>
+        /// Explicitly unregisters all events related to the two buttons in OnApplyTemplate().
+        /// </summary>
+        private void UnregisterEvents()
+        {
+            if (_primaryButton != null)
+            {
+                _primaryButton.Click -= PrimaryButton_Click;
+            }
+
+            if (_secondaryButton != null)
+            {
+                _secondaryButton.Click -= SecondaryButton_Click;
+            }
+        }
+
+        ////////////////////////////////////////////////////////////////////////
+        // OnEvent Overridable Methods
+        ////////////////////////////////////////////////////////////////////////
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate(TemplateAppliedEventArgs e)
+        {
+            UnregisterEvents();
+            UnregisterFlyoutEvents(Flyout);
+
+            _primaryButton   = e.NameScope.Find<Button>("PART_PrimaryButton");
+            _secondaryButton = e.NameScope.Find<Button>("PART_SecondaryButton");
+
+            if (_primaryButton != null)
+            {
+                _primaryButton.Click += PrimaryButton_Click;
+            }
```

Comment:
Removing the pointer events and using the underlying button's pseudoclasses it already sets is a nice idea. That is a heavy simplification here in code behind.